### PR TITLE
feat: Set token required in Tokenless section under org upload token 

### PR DESCRIPTION
**Pull request by RulaKhaled**
# Description
Hook up API work with tokenless section under org upload token tab.

depends on:
- [x] #3358 

# Notable Changes
- New mutation hook 
- Hook up api work with section 
- A lot of tests 

# Screenshots
Nothing visual changed, notification on a successful mutation is sent:

https://github.com/user-attachments/assets/62b49724-f023-4300-9238-fcd6634cba23

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.test.tsx [hunk 1]
```diff
@@ -1,139 +1,241 @@
...
+vi.mock('shared/featureFlags')
+vi.mock('services/toastNotification', async () => {
+  const actual = await vi.importActual('services/toastNotification')
+  return {
+    ...actual,
+    useAddNotification: mocks.useAddNotification,
+  }
+})
+
+const mockedUseFlags = useFlags as jest.Mock
```
Comment (ajay-sentry) [inline src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.test.tsx:26]: I think we may need to swap these to vi.mock() similar to this

https://github.com/codecov/gazebo/blob/01a7a7cf16d3b19e6b9e7f34096c0120ca6aed51/src/pages/AnalyticsPage/Tabs/Tabs.test.tsx#L21-L22

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.tsx [hunk 2]
```diff
@@ -1,25 +1,52 @@
...
+    useSetUploadTokenRequired()
+
+  const [showModal, setShowModal] = useState<boolean>(false)
+  const [tokenRequired, setTokenRequiredState] = useState<boolean>(true)
+
+  useEffect(() => {
+    if (
+      !isUploadTokenRequiredLoading &&
+      uploadTokenRequiredData &&
+      uploadTokenRequiredData?.uploadTokenRequired !== null
```
Comment (ajay-sentry) [inline src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.tsx:38]: nit: if we're checking for uploadTokenRequiredData I think we don't need the ? on uploadTokenRequiredData?.uploadTokenRequired

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/useSetUploadTokenRequired.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        })
+      } else {
+        addToast({
+          type: 'success',
+          text: 'Upload token requirement updated successfully',
+          disappearAfter: TOAST_DURATION,
+        })
+      }
+    },
+    onError: () => {
```
Comment (ajay-sentry) [inline src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/useSetUploadTokenRequired.tsx:69]: weird how it says this line is uncovered when you specifically have a test for it 🤔

**Review by ajay-sentry — APPROVED** [review]
couple small comments but nothing blocking

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.test.tsx (hunk 1 above)
Comment (RulaKhaled) [reply src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.test.tsx:26]: good catch

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/useSetUploadTokenRequired.tsx (hunk 3 above)
Comment (RulaKhaled) [reply src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/useSetUploadTokenRequired.tsx:69]: I KNOW RIGHT

Comment (RulaKhaled) [reply src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/useSetUploadTokenRequired.tsx:69]: i have some feedback for codecov..

File: src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.tsx (hunk 2 above)
Comment (RulaKhaled) [reply src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.tsx:38]: TS is weird, this actually tests if it's set to boolean, if i just do uploadTokenRequiredData?.uploadTokenRequired !== null and set it to `uploadTokenRequiredData?.uploadTokenRequired` it complains about a nullable value: 
Screenshot 2024-10-10 at 5 29 32 AM
Screenshot 2024-10-10 at 5 29 29 AM

**Pull request merged**